core/src/bitmap/bitmap_data.rs

```diff
@@ -1277,6 +1277,40 @@ impl<'gc> BitmapData<'gc> {
         self.avm2_object = Some(object)
     }
 
+    pub fn apply_filter(
+        &mut self,
+        context: &mut UpdateContext<'_, 'gc>,
+        source: BitmapHandle,
+        source_point: (u32, u32),
+        source_size: (u32, u32),
+        dest_point: (u32, u32),
+        filter: Filter,
+    ) {
+        let dest = self.bitmap_handle(context.renderer).unwrap();
+
+        self.update_dirty_texture(context.renderer);
```

Comment:
This is almost identical to the BitmapData.draw implementation. I think it could be factored out into a method that takes a `FnOnce() -> Option<SyncHandle>` closure, which calls `update_dirty_texture`, calls the closure, and then processes the `SyncHandle`.

Reply:
Good idea

Reply:
@Aaron1011 Tried this and it's not that easy, we need to mutably borrow the renderer but we can't do that if the closure needs it too (which it will). Could probably make a function of the `match self.dirty_state` but that's about it